mesonbuild/backend/ninjabackend.py

```diff
@@ -2336,6 +2336,12 @@ def determine_rpath_dirs(self, target):
     def get_dependency_filename(self, t):
         if isinstance(t, build.SharedLibrary):
             return os.path.join(self.get_target_private_dir(t), self.get_target_filename(t) + '.symbols')
+        elif isinstance(t, mesonlib.File):
+            if t.is_built:
+                return t.relative_name()
+            else:
+                return t.absolute_path(self.environment.get_source_dir(),
+                                       self.environment.get_build_dir())
```

Comment:
It gets passed to `self.get_target_filename()` if it isn't handled in that if block, which is what it did before my patch. Is that not correct?